moveit_py/src/moveit/moveit_py_utils/src/copy_ros_msg.cpp

```diff
@@ -0,0 +1,362 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2022, Peter David Fagan
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Peter David Fagan */
+
+#include <list>
+#include <moveit_py/moveit_py_utils/copy_ros_msg.h>
+
+namespace moveit_py
+{
+namespace moveit_py_utils
+{
+// Ros Message Copy Definitions (Note: copying faster than serialize/deserialize)
+
+geometry_msgs::msg::PoseStamped PoseStampedToCpp(const py::object& pose_stamped)
+{
+  // recreate instance in C++ using python object data
+  geometry_msgs::msg::PoseStamped pose_stamped_cpp;
+  pose_stamped_cpp.header.frame_id = pose_stamped.attr("header").attr("frame_id").cast<std::string>();
+  pose_stamped_cpp.pose.orientation.w = pose_stamped.attr("pose").attr("orientation").attr("w").cast<double>();
+  pose_stamped_cpp.pose.position.x = pose_stamped.attr("pose").attr("position").attr("x").cast<double>();
+  pose_stamped_cpp.pose.position.y = pose_stamped.attr("pose").attr("position").attr("y").cast<double>();
+  pose_stamped_cpp.pose.position.z = pose_stamped.attr("pose").attr("position").attr("z").cast<double>();
```

Comment:
This seems wrong.

Don't we want to copy orientation (w, x, y, z) and position (x, y, z)? Same with the other functions.

Also, is there an opportunity for this function to just copy the stamp and then directly call `PoseToCpp` below? Seems less repetitive.

Reply:
This is a severe bug, fixing now.

Reply:
[2e40610](https://github.com/ros-planning/moveit2/pull/1546/commits/2e406103c24f58003445a3693bdb79e136e1cbcc)